src/io/index.js

```diff
@@ -0,0 +1,308 @@
+        });
+    }
+
+    // Creates a file from an encoded data uri
+    async createFileFromEncodedDataURI(path) {
+        return new Promise((resolve, reject) => {
+            var dataURL = path.substring(path.indexOf('data:'));
+            var blob = this.dataURLtoBlob(dataURL);
+            var paths = path.substring(0, path.indexOf('data:') - 1);
+            var fileName = pth.basename(paths);
+            var filePath = pth.dirname(paths) + '/';
+
+            const handleUpload = async blob => {
+                try {
+                    const fileContents = await this.readUploadedFileAsArrayBuffer(
+                        blob
+                    );
+                    var buf = new Int8Array(fileContents);
+                    var file = { name: fileName, path: filePath, buffer: buf };
+                    const result = await this.createFileFromArrayBuffer(file);
+                    result.path = filePath;
+                    resolve(result);
+                } catch (e) {
+                    return reject(e.message);
+                }
+            };
+            handleUpload(blob);
+        });
+    }
+
+    // Generates a Data URI for the specified file
+    async getFileDataURL(path) {
+        return new Promise((resolve, reject) => {
+            const handle = async () => {
+                try {
+                    const fileInfo = await this.getFileInfo(path);
+                    var contents = fileInfo.body.contents;
+                    var name = fileInfo.body.name;
+                    var file = new File([contents], name, {
+                        type: fileInfo.type,
+                    });
+                    const fileDataURI = await this.readUploadedFileAsDataURL(
+                        file
+                    );
+                    resolve({ name: name, dataurl: fileDataURI });
+                } catch (e) {
+                    return reject(e.message);
+                }
+            };
+            handle();
+        });
+    }
+
+    // Uploads a file into filesystem
+    async createFileFromArrayBuffer(file) {
+        return new Promise((resolve, reject) => {
+            var buf = buffer(file.buffer);
+            fs.writeFile(file.path + file.name, buf, function(err) {
+                if (err) return reject({ success: false, err: err });
+                else resolve({ success: true });
+            });
+        });
+    }
+
+    // Uploads array of files into filesystem
+    async createFilesFromArrayBuffer(files) {
+        const promises = files.map(async file => {
+            file.buffer = Object.values(file.buffer);
+            const result = await this.createFileFromArrayBuffer(file);
+            return result;
+        });
+
+        return await Promise.all(promises);
+    }
+
+    // Deletes everything in Path
+    async deletePathRecursively(path) {
+        return new Promise((resolve, reject) => {
+            fs.stat(path, function(err, stats) {
```

Comment:
If you use `sh.rm` it will do this for you automatically, and then you don't have to stat and deal with file vs. folder manually.